src/services/document_service.py

```diff
@@ -183,39 +183,48 @@ def save(
                 node.get_context_storage_attribute(),
                 parent_id=parent_uid,
             )
-            return {"type": SIMOS.REFERENCE.value, "address": node.uid, "referenceType": REFERENCE_TYPES.LINK.value}
+            result = {
+                "type": SIMOS.REFERENCE.value,
+                "address": f"${node.uid}",
+                "referenceType": REFERENCE_TYPES.LINK.value,
+            }
+            return result
         return ref_dict
 
-    def get_document_by_uid(
-        self,
-        data_source_id: str,
-        document_uid: str,
-        depth: int = 999,
-    ) -> dict:
-        data_source = self.get_data_source(data_source_id)
-        document = data_source.get(document_uid)
-        return resolve_document(document, data_source, self.get_data_source, document_uid, depth)
-
-    def get_node_by_uid(self, data_source_id: str, document_uid: str, depth: int = 999) -> Node:
-        complete_document = self.get_document_by_uid(data_source_id, document_uid, depth)
-        return tree_node_from_dict(
-            complete_document,
-            uid=complete_document.get("_id"),
+    def get_document(self, reference: str, depth: int = 999) -> Node:
+        resolved_reference: ResolvedReference = resolve_reference(reference, self.get_data_source)
+
+        data_source: DataSource = self.get_data_source(resolved_reference.data_source_id)
+        document: dict = data_source.get(resolved_reference.document_id)
+
+        resolved_document: dict = resolve_document(
+            document,
+            data_source,
+            self.get_data_source,
+            resolved_reference.document_id,
+            depth + len(resolved_reference.attribute_path.split(".")),
+        )
+
```

Comment:
can add check on depth here and return resolved document if depth is 0 (or 1?? not sure what "indexing scheme" we use for depth - 0 or 1 indexing)

Reply:
created a sperate task in the issue for this